core/src/main/java/com/linecorp/armeria/server/logging/AccessLogComponent.java

```diff
@@ -93,16 +99,101 @@ public Object getMessage(RequestLog log) {
         }
     }
 
+    /**
+     * A timestamp component of a log message.
+     */
+    class TimestampComponent implements AccessLogComponent {
+
+        @VisibleForTesting
+        static final DateTimeFormatter defaultDateTimeFormatter =
+                DateTimeFormatter.ofPattern("dd/MMM/yyyy:HH:mm:ss Z", Locale.ENGLISH);
+        @VisibleForTesting
+        static final ZoneId defaultZoneId = ZoneId.systemDefault();
+
+        static boolean isSupported(AccessLogType type) {
+            return type == AccessLogType.REQUEST_TIMESTAMP;
+        }
+
+        private final boolean addQuote;
+        private final DateTimeFormatter formatter;
+
+        TimestampComponent(boolean addQuote, @Nullable String variable) {
+            this.addQuote = addQuote;
+            formatter = findFormatter(variable);
+        }
+
+        @Nullable
+        @Override
+        public Object getMessage(RequestLog log) {
+            return formatter.format(ZonedDateTime.ofInstant(
+                    Instant.ofEpochMilli(log.requestStartTimeMillis()), defaultZoneId));
+        }
+
+        @Override
+        public boolean addQuote() {
+            return addQuote;
+        }
+
+        static DateTimeFormatter findFormatter(@Nullable String variable) {
+            if (variable == null) {
+                return defaultDateTimeFormatter;
+            }
+            // Find pre-defined formatter first.
+            switch (variable) {
+                case "BASIC_ISO_DATE":
+                    return DateTimeFormatter.BASIC_ISO_DATE;
```

Comment:
Perhaps we could pre-populate `Map<String, DateTimeFormatter>` by looking for the `public static final DateTimeFormatter` fields in `DataTimeFormatter`?

Reply:
Got it!